zipline/data/data_portal.py

```diff
@@ -1360,24 +1363,13 @@ def get_fetcher_assets(self, day):
         if self._extra_source_df is None:
             return []
 
-        if day in self._extra_source_df.index:
-            date_to_use = day
-        else:
-            # current day isn't in the fetcher df, go back the last
-            # available day
-            idx = self._extra_source_df.index.searchsorted(day)
-            if idx == 0:
-                return []
-
-            date_to_use = self._extra_source_df.index[idx - 1]
-
-        asset_list = self._extra_source_df.loc[date_to_use]["sid"]
+        day = normalize_date(dt)
 
-        # make sure they're actually assets
-        asset_list = [asset for asset in asset_list
-                      if isinstance(asset, Asset)]
-
-        return asset_list
+        assets = self._extra_source_df.loc[day]['sid']
+        if isinstance(assets, pd.Series):
```

Comment:
I'm not that familiar with pandas, but I noticed that if `self._extra_source_df` only has 1 row with the indexed `day`, then `self._extra_source_df.loc[day]` returns a Series and then `self._extra_source_df.loc[day]['sid']` returns a scalar. Only if there are multiple rows with that `day` does `self._extra_source_df.loc[day]['sid']` return a Series